src/components/select/utils/with-filter.hoc.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import I18n from 'i18n-js';
+import filterChildren from '../../../utils/filter-children';
+import StyledOption from '../option/option.style';
+import highlightPartOfText from './highlight-part-of-text';
+
+/** Filters wrapped component children based on provided filter text and highlights matching content */
+const withFilter = (WrappedComponent) => {
+  const FilteredComponent = ({
+    children,
+    filterText,
+    forwardRef,
```

Comment:
Please can we use `React.forwardRef` instead so we can use the regular `ref` prop.

Reply:
Unresolved.

Reply:
there was a forward ref on the bottom of the file, moved it to the top for clarity